src/vs/editor/contrib/suggest/suggestWidget.ts

```diff
@@ -456,7 +463,8 @@ export class SuggestWidget implements IContentWidget, IVirtualDelegate<ICompleti
 				listInactiveFocusOutline: activeContrastBorder
 			}),
 			themeService.onThemeChange(t => this.onThemeChange(t)),
-			editor.onDidBlurEditorText(() => this.onEditorBlur()),
+			editor.onDidFocusEditorText(() => this.onEditorTextEvent()),
```

Comment:
Why are we adding a listener for editor focus?

Reply:
We have a listener for the editor blur which closes the suggest widget (and the details element get closed as well). Since we don't want to close the widget when the user clicks on the detials element, I added a flag to know when it is focused (meaning the mouse enters the element).

When the user clicks (and/or select text) on the detail element, the editor blur listener was already called and we prevented the widget to be closed because the details element was focused, so, we needed a way to close the widget when the user focus the editor back. 

That's why we added the listener for editor focus.

Reply:
Take this case:
- User clicks on the detail element
- User clicks anywhere but the editor (menus, sidebar, outside vscode)

Now we end up with the suggest widget not disappearing at all.

Reply:
What you need is blur event on the details. When that event is fired, we should close the widget.

Reply:
We have the same behavior already, look:

**Before my changes:**
![vscode-before](https://user-images.githubusercontent.com/4671080/43934939-a892ef7a-9c1f-11e8-9d19-d4da23442dd8.gif)

**After my changes:**
![vscode-after](https://user-images.githubusercontent.com/4671080/43934952-bb06af3e-9c1f-11e8-9b79-8e7feb76dbcd.gif)

If the user clicks sidebar, console, editor and others parts it gets closed. Scrolling or clicking outside vscode doesn't close it though.

Reply:
I'm not sure that adding the editor blur listener to details will fix the issue since clicking on details means losing focus from the editor and then the widget will get closed (the same behavior we are trying to fix).

So, to be on the same page we want to no add the editor focus listener,  as suggested [here](https://github.com/Microsoft/vscode/pull/56052#discussion_r209117794), right?

Also, maybe what we want is to close the widget on `mouseleave` as the hover widget does?

Reply:
> If the user clicks sidebar, console, editor and others parts it gets closed. Scrolling or clicking outside vscode doesn't close it though.

With your changes, when I click on the sidebar/console after selecting text from the details, the widget doesnt close.

On a side note, let's wait till #56123 gets merged which will fix the feature `sticky` was meant for.

Reply:
ok, great. I'll keep an eye on it.

Reply:
@wistcc #56123 is merged. Can you get the latest from master, and then see how your changes are affected by it?

Reply:
Cool, on it.

I'm pretty sure I will need to change my logic. I'll let you know when it's ready for another request. Thanks for the heads up!